process/block/sovereign/outgoingOperations_test.go

```diff
@@ -25,115 +25,376 @@ func createEvents() []SubscribedEvent {
 	t.Parallel()
 
 	t.Run("no subscribed events, should return error", func(t *testing.T) {
-		creator, err := NewOutgoingOperationsFormatter([]SubscribedEvent{}, &testscommon.RoundHandlerMock{}, &mock.DataCodecMock{})
+		t.Parallel()
+
+		args := createArgs()
+		args.SubscribedEvents = []SubscribedEvent{}
+		creator, err := NewOutgoingOperationsFormatter(args)
 		require.Nil(t, creator)
 		require.Equal(t, errNoSubscribedEvent, err)
 	})
 
-	t.Run("nil round handler, should return error", func(t *testing.T) {
-		events := createEvents()
-		creator, err := NewOutgoingOperationsFormatter(events, nil, &mock.DataCodecMock{})
+	t.Run("nil data codec, should return error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createArgs()
+		args.DataCodec = nil
+		creator, err := NewOutgoingOperationsFormatter(args)
 		require.Nil(t, creator)
-		require.Equal(t, errors.ErrNilRoundHandler, err)
+		require.Equal(t, errors.ErrNilDataCodec, err)
 	})
 
-	t.Run("nil data codec, should return error", func(t *testing.T) {
-		events := createEvents()
-		creator, err := NewOutgoingOperationsFormatter(events, &testscommon.RoundHandlerMock{}, nil)
+	t.Run("nil topics checker, should return error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createArgs()
+		args.TopicsChecker = nil
+		creator, err := NewOutgoingOperationsFormatter(args)
 		require.Nil(t, creator)
-		require.Equal(t, errors.ErrNilDataCodec, err)
+		require.Equal(t, errors.ErrNilTopicsChecker, err)
 	})
 
 	t.Run("should work", func(t *testing.T) {
-		events := createEvents()
-		creator, err := NewOutgoingOperationsFormatter(events, &testscommon.RoundHandlerMock{}, &mock.DataCodecMock{})
+		t.Parallel()
+
+		args := createArgs()
+		creator, err := NewOutgoingOperationsFormatter(args)
 		require.Nil(t, err)
 		require.False(t, creator.IsInterfaceNil())
 	})
 }
 
-func TestOutgoingOperations_CreateOutgoingTxData(t *testing.T) {
+func createValidArgs() *outgoingOperations {
+	dataCodec := &mock.DataCodecMock{}
+	topicsChecker := &mock.TopicsCheckerMock{
+		CheckValidityCalled: func(topics [][]byte) error {
+			return nil
+		},
+	}
+
+	events := []SubscribedEvent{
+		{
+			Identifier: []byte("deposit"),
+			Addresses: map[string]string{
+				"addr1": "addr1",
+				"addr2": "addr2",
+			},
+		},
+	}
+
+	args := ArgsOutgoingOperations{
+		SubscribedEvents: events,
+		DataCodec:        dataCodec,
+		TopicsChecker:    topicsChecker,
+	}
+	creator, _ := NewOutgoingOperationsFormatter(args)
+	return creator
+}
+
+func TestOutgoingOperations_CheckEvent(t *testing.T) {
 	t.Parallel()
 
-	addr1 := []byte("addr1")
-	addr2 := []byte("addr2")
-	addr3 := []byte("addr3")
+	t.Run("invalid identifier", func(t *testing.T) {
+		t.Parallel()
 
-	identifier1 := []byte("deposit")
-	identifier2 := []byte("send")
+		events := []SubscribedEvent{
+			{
+				Identifier: []byte(""),
+				Addresses: map[string]string{
+					"addr1": "addr1",
+				},
+			},
+		}
+
+		args := ArgsOutgoingOperations{
```

Comment:
Here and below you could use func to create args and only overwrite the subscribed events